mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/MavenPropertyResolver.java

```diff
@@ -1,32 +1,94 @@
 package org.arquillian.smart.testing.mvn.ext;
 
 import java.util.Arrays;
+import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
 
 class MavenPropertyResolver {
 
+    private final Model model;
+    private final Plugin surefirePlugin;
+    private final Plugin failsafePlugin;
+
     private static final Pattern TEST_CLASS_PATTERN = Pattern.compile("[^a-z0-9 ]", Pattern.CASE_INSENSITIVE);
 
+    MavenPropertyResolver(Model model) {
+        this.model = model;
+        this.surefirePlugin = getPlugin("maven-surefire-plugin");
+        this.failsafePlugin = getPlugin("maven-failsafe-plugin");
```

Comment:
Don't we have something similar in place already? https://github.com/arquillian/smart-testing/blob/master/mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/ApplicablePlugins.java